worker_test.go

```
@@ -576,3 +576,115 @@ func TestWorkerGroup(t *testing.T) {
+		w := newWorker(func(data string) {})
+
+		// Create a worker binder
+		binder := newQueues(w)
+
+		// Test BindQueue
+		t.Run("BindQueue", func(t *testing.T) {
+			queue := binder.BindQueue()
+			assert.NotNil(t, queue, "Queue should not be nil")
+
+			// Check if returned type implements Queue interface
+			_, ok := queue.(Queue[string])
```

golangci-lint (1.64.8)

S1040: type assertion to the same type: queue already has type Queue[string]

(gosimple)